cloudflare.php

```diff
@@ -46,4 +46,6 @@
 
 // Plugin uses namespaces. To support old PHP version which doesn't support
 // namespaces we load everything in "cloudflare.loader.php"
-require_once CLOUDFLARE_PLUGIN_DIR.'cloudflare.loader.php';
+add_action('init', function () {
```

Comment:
At the end of the day, this should probably be `plugins_loaded`. 

1. In general, we shouldn't be aiming for compatibility for theme-based overrides in plugins - rather optimize for recommending mu-plugins.

2. Theme-based override compatibility can also be achieved by simply not doing "too early for themes" things in the loader, but that probably requires extra refactoring. Workload vs benefits question.

Reply:
I think a lot of people try to alter hooks from a theme. It's pretty rare that people use so called site plugin to change functionality and just use a theme just for styling.

In my eyes, this is critical to make it work within themes.

What's the issue in delaying the initiation of the full Cloudflare loader?

Reply:
> What's the issue in delaying the initiation of the full Cloudflare loader?

It's true, in general, `init` should be fine, too. But https://github.com/cloudflare/Cloudflare-WordPress/pull/411#issuecomment-851162263 seems indicate it might be too late, so perhaps additional refactoring is already required with this move.

Reply:
Loading it at `init` works for my use case. Adding a filter for `cloudflare_purge_url_actions` in the theme.

Reply:
Can this be reviewed? It may not solve all of the issues but it will solve some.